dmt/A_mappers/ros_A_mapper.py

```diff
@@ -230,24 +230,61 @@ def find_message_module(msg: str) -> str:
 
 
 def generate_basic_type_message(msg: str) -> dict:
-    try:
-        definition = asnParser.g_names[msg]
-    except KeyError:
+    if msg not in asnParser.g_names:
         panic("I don't have the info to build message %s" % (msg))
+    definition = asnParser.g_names[msg]
 
     msg_dict = {}
     msg_dict['constants'] = []
     msg_dict['variables'] = []
     try:
         msg_dict['variables'].append(
             [find_type(definition), 'data', definition._range[1]])
-        print(definition._range[1])
     except AttributeError:
         msg_dict['variables'].append(
             [find_type(definition), 'data'])
     return msg_dict
 
 
+def get_variable_data(asnParser, member_type, leaf_type):
+    if leaf_type is 'CHOICE':
+        panic("ROS doesn't support choice messages")
+    elif leaf_type is 'ENUMERATED':
+        enum_members = asnParser.g_names[member[1]._leafType]._members
+        data_type = get_enum_data_type(enum_members)
+        return [data_type, member[0]]
+    elif member_type in primitive_types.keys():
+        return [primitive_types[member_type], member[0]]
```

Comment:
Which `member` is this referring to? Is that a global?

Reply:
That was a mistake in the function args. Now is fixed ;)